$MirOS$
$OpenBSD: patch-jerror_c,v 1.3 2013/04/20 15:23:40 naddy Exp $

This one is slightly problematic.  If an application allocates less
room for its error buffer than the recommended JMSG_LENGTH_MAX, the
error message buffer will still overflow.

--- jerror.c.orig	Sun Feb 22 01:02:54 1998
+++ jerror.c	Sun Aug  4 12:22:47 2013
@@ -190,9 +190,9 @@ format_message (j_common_ptr cinfo, char
 
   /* Format the message into the passed buffer */
   if (isstring)
-    sprintf(buffer, msgtext, err->msg_parm.s);
+    snprintf(buffer, JMSG_LENGTH_MAX, msgtext, err->msg_parm.s);
   else
-    sprintf(buffer, msgtext,
+    snprintf(buffer, JMSG_LENGTH_MAX, msgtext,
 	    err->msg_parm.i[0], err->msg_parm.i[1],
 	    err->msg_parm.i[2], err->msg_parm.i[3],
 	    err->msg_parm.i[4], err->msg_parm.i[5],
